workflow/sync/multi_throttler.go

```diff
@@ -139,6 +139,10 @@ func (m *multiThrottler) Remove(key Key) {
 
 	namespace, _, _ := cache.SplitMetaNamespaceKey(key)
 	delete(m.running, key)
+	_, ok := m.pending[namespace]
+	if !ok {
```

Comment:
We could just skip the remove instead? I don't mind this as is, but it reads weird to call remove on a map we've just created.